synapse/handlers/receipts.py

```diff
@@ -165,43 +165,31 @@ def __init__(self, hs: "HomeServer"):
         self.config = hs.config
 
     @staticmethod
-    def filter_out_private(events: List[JsonDict], user_id: str) -> List[JsonDict]:
+    def filter_out_private(events: List[JsonDict], our_user_id: str) -> List[JsonDict]:
         """
         This method takes in what is returned by
         get_linearized_receipts_for_rooms() and goes through read receipts
         filtering out m.read.private receipts if they were not sent by the
         current user.
         """
 
-        visible_events = []
-
         # filter out private receipts the user shouldn't see
-        for event in events:
+        for index, event in enumerate(events):
             content = event.get("content", {})
-            new_event = event.copy()
```

Comment:
I think we still need to copy this in order to avoid mutating what's in the cache? (It might be worth writing a test for this actually: have a user without the hidden read receipt request it, then have the user with it request it and ensure it is still there. We might already have one which covers it, not sure!)

Reply:
Ah, true